File: src/main/java/me/aov/sellgui/api/SellGUIItemsSoldEvent.java
package me.aov.sellgui.api;

import org.bukkit.entity.Player;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

// soon lol
public class SellGUIItemsSoldEvent extends Event {
    private static final HandlerList handlers = new HandlerList();
    private final Player player;
    private final double totalSoldValue;

    public SellGUIItemsSoldEvent(Player player, double totalSoldValue) {
        this.player = player;
        this.totalSoldValue = totalSoldValue;
    }

    public Player getPlayer() {
        return player;
    }

    public double getTotalSoldValue() {
        return totalSoldValue;
    }

    @Override
    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/main/kotlin/me/aov/sellgui/api/SellGUIItemsSoldEvent.kt
package me.aov.sellgui.api

import org.bukkit.entity.Player
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class SellGUIItemsSoldEvent(private val player: Player, private val totalSoldValue: Double) : Event() {
    fun getPlayer(): Player = player

    fun getTotalSoldValue(): Double = totalSoldValue

    override fun getHandlers(): HandlerList = handlers

    companion object {
        private val handlers = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList = handlers
    }
}
